vespalib/src/vespa/vespalib/net/tls/tls_context.h

```diff
@@ -6,11 +6,21 @@
 namespace vespalib::net::tls {
 
 class TransportSecurityOptions;
+class CertificateVerificationCallback;
 
 struct TlsContext {
     virtual ~TlsContext() = default;
 
+    // Create a TLS context which verifies certificates according to the provided options'
+    // CA trust roots AND allowed peer policies
     static std::shared_ptr<TlsContext> create_default_context(const TransportSecurityOptions&);
+    // Create a TLS context where the certificate verification callback is explicitly provided.
+    // IMPORTANT: This does NOT verify that the peer satisfies the allowed peer policies!
+    //            It only verifies that a peer is signed by a trusted CA. This function should
+    //            therefore only be used in very special circumstances, such as unit tests.
+    static std::shared_ptr<TlsContext> create_default_context(
+            const TransportSecurityOptions&,
+            std::shared_ptr<CertificateVerificationCallback> cert_verify_callback);
```

Comment:
alternative solution: simple AND between the custom verifier and the one created from the allowed_peers specification.

Reply:
Right now this is only used by tests that explicitly want to control the callback, but if we want to make this functionality more general I agree that a composite callback is the way to go. Will consider for a follow-up.